From a0943bd277ed9b5ecaf1c7c1ace7590b1b6da37d Mon Sep 17 00:00:00 2001
From: Shilin Victor <chrono.monochrome@gmail.com>
Date: Mon, 20 Aug 2018 22:57:43 +0300
Subject: [PATCH] modify overscroll bug in NotificationStackScrollLayout After overscroll, 
notifiation panel is seen as being cut. Initialize overscroll when you unlock

diff --git a/packages/SystemUI/src/com/android/systemui/statusbar/stack/NotificationStackScrollLayout.java b/packages/SystemUI/src/com/android/systemui/statusbar/stack/NotificationStackScrollLayout.java
index 75532d9..4dbe4df 100644
--- a/packages/SystemUI/src/com/android/systemui/statusbar/stack/NotificationStackScrollLayout.java
+++ b/packages/SystemUI/src/com/android/systemui/statusbar/stack/NotificationStackScrollLayout.java
@@ -4284,6 +4284,9 @@
     }
 
     public void setStatusBarState(int statusBarState) {
+        if (mStatusBarState != statusBarState) {
+            endDrag();
+        }
         mStatusBarState = statusBarState;
         mAmbientState.setStatusBarState(statusBarState);
     }
